src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -409,17 +412,67 @@ def _process_heatmap_tools(heatmap_tools: set[str],
 
-        if bad_data_ids:
-            result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool=tool)
-            with open(os.path.join(result_dir, BAD_DATA_IDS), "w") as outfile:
-                json.dump(bad_data_ids, outfile)
 
+def _process_fatal_error_tools(check_fatal_error_tools: set[str],
+                                root_dir: str,
+                                kbase_collection: str,
+                                load_ver: str):
+    if not check_fatal_error_tools:
+        return set()
+    
+    fatal_dict = dict()
+    for tool in check_fatal_error_tools:
+        result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool=tool)
+        batch_dirs = _get_batch_dirs(result_dir)
+        if len(batch_dirs) == 1 and batch_dirs[0].startswith(loader_common_names.COMPUTE_OUTPUT_NO_BATCH_PREFIX):
+            batch_dirs = [os.path.join(batch_dirs[0], d) for d in os.listdir(batch_dirs[0])
+                          if os.path.isdir(os.path.join(result_dir, batch_dirs[0], d))]
+        for batch_dir in batch_dirs:
+            data_dir = os.path.join(result_dir, batch_dir)
+            fatal_error_file = os.path.join(data_dir, loader_common_names.FATAL_ERROR_FILE)
+            if not os.path.exists(fatal_error_file):
+                continue
+            try:
+                with open(fatal_error_file, "r") as json_file:
+                    fatal_errors = json.load(json_file)
+            except Exception as e:
+                raise ValueError(f"{fatal_error_file} exists, but unable to retrive") from e
+            
+            for kbase_id in fatal_errors:
+                fatal_dict_info = loader_helper.create_global_fatal_dict_doc(
+                    tool, fatal_errors[kbase_id][loader_common_names.ERROR], fatal_errors[kbase_id][loader_common_names.STACKTRACE])
+                if fatal_dict.get(kbase_id):
+                    fatal_dict[kbase_id][loader_common_names.ERRORS].append(
+                        fatal_dict_info)
+                else:
+                    fatal_dict[kbase_id] = {loader_common_names.FILE: fatal_errors[kbase_id][loader_common_names.FILE],
+                                            loader_common_names.ERRORS: [fatal_dict_info]}
+    
+    import_dir = os.path.join(root_dir, IMPORT_DIR)
+    os.makedirs(import_dir, exist_ok=True)
+    fatal_output = f"{kbase_collection}_{load_ver}_{loader_common_names.FATAL_ERROR_FILE}"
+    fatal_error_path = os.path.join(import_dir, fatal_output)
+    print(f"Creating a merged {loader_common_names.FATAL_ERROR_FILE}: {fatal_error_path}")
+    with open(fatal_error_path, "w") as outfile:
+        outfile.dump(fatal_dict, outfile)
+    
+    return set(fatal_dict.keys())
+
+
+def _remove_docs_with_fatal_ids(docs, fatal_ids):
+    res = []
+    for doc in docs:
+        if doc[names.FLD_KBASE_ID] not in fatal_ids:
+            res.append(doc)
+    return res
```

Comment:
```suggestion
   return [d for d in docs if d[names.FLD_KBASE_ID] not in fatal_ids]
```

Reply:
Updated! We'd avoid processing the record at all if it's in fatal_ids set.

Reply:
I'm not sure I understand - aren't these equivalent implementations of the logic?

Reply:
Yes, they are. But we don't need this function anymore. It was used after merge_docs step. Now the logic is we'd avoid processing the record at all.

Reply:
Ok great

Reply:
:+1: